.github/workflows/sphinx-python-dependency-build-checks.yml

```diff
@@ -0,0 +1,49 @@
+# The purpose of this workflow file is to confirm that the Sphinx
+# virtual environment can be built from source, consequently documenting
+# the packages required in the build environment to do that.
+#
+# This is needed because some projects embeds the documentation into built
+# artifacts which involves rendering the documentation on the target
+# architecture.
+#
+# Depending on the architecture, pip may or may not have already built wheels
+# available, and as such we need to make sure building wheels from source can
+# succeed.
+name: Check and document build requirements for Sphinx venv
+on:
+  - push
+  - pull_request
+  - workflow_dispatch
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true  
+
+jobs:
```

Comment:
How can I run those checks locally if the code is only in a github workflow? previously we used tox to build & test the documentation. and then call tox from a github workflow. what do we want to change that?

Reply:
`make` provides full help text. `make html` is generally the most important, but there are targets for links, spelling, and other linting issues too.